.github/workflows/deps-update.yml

```diff
@@ -0,0 +1,116 @@
+name: Dependency Updates
+
+on:
+  schedule:
+    - cron: '42 5 * * 1' # Mondays at 5:42 AM
+  workflow_dispatch: {} # Manual runs
+
+permissions: read-all
+
+jobs:
+  update-go:
+    name: Update Go Dependencies
+    runs-on: ubuntu-latest
+    outputs:
+      changes-needed: ${{ steps.is-tree-dirty.outputs.result }}
+    env:
+      GOTOOLCHAIN: local # Prohibits adding `toolchain` directives to go.mod files.
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
+
+      - name: Set up Go
+        uses: actions/setup-go@0a12ed9d6a96ab950c8f026ed9f722fe0da7ef32 # v5
+        with:
+          go-version: '1.21' # Minimum supported go release
+          cache-dependency-path: '**/*.sum'
+
+      # Select the latest available version of gopkg.in/DataDog/dd-trace-go.v1, while ignoring all
+      # the `v1.999.*` versions, which are experimental pre-releases.
+      - name: Determine gopkg.in/DataDog/dd-trace-go.v1 version
+        id: dd-trace-go
+        run: |-
+          set -euo pipefail
+          version=$(go list -f '{{range .Versions}}{{.}}{{"\n"}}{{end}}' -m -versions gopkg.in/DataDog/dd-trace-go.v1 | grep -v -E '^v1\.999\.' | tail -n 1)
```

Comment:
Are we confident enough about the version ordering to be able to use ˋtail -n 1` ?

Reply:
According to `go list` documentation:

>The `-versions` flag causes list to set the Module's Versions field
to a list of all known versions of that module, ordered according
to semantic versioning, earliest to latest.